pkg/cloudprovider/events/events.go

```diff
@@ -40,3 +46,22 @@ func NodeClaimFailedToResolveNodeClass(nodeClaim *v1.NodeClaim) events.Event {
 		DedupeValues:   []string{string(nodeClaim.UID)},
 	}
 }
+
+func NodeClaimFailedToRegister(nodeClaim *v1.NodeClaim, err error) events.Event {
+	return events.Event{
+		InvolvedObject: nodeClaim,
+		Type:           corev1.EventTypeWarning,
+		Reason:         AsyncProvisioningReason, // TODO: our other events don't set reason... is it an oversight?
```

Comment:
No reason elsewhere: Yes, oversight

Reply:
I have removed this comment. Do you want me to just add a reason to the other 2 events here?

Reply:
I've added a reason here too.